examples_apps/run_state_learner.py

```diff
@@ -0,0 +1,432 @@
+
+# Create the Strawberry Fields free parameters for each layer
+sf_params = []
+names = ["r1", "sq_r", "sq_phi", "r2", "d_r", "d_phi", "kappa"]
+
+for k in range(depth):
+    sf_params_names = ["{}_{}".format(n, k) for n in names]
+    sf_params.append(prog.params(*sf_params_names))
+
+######################################################################
+# Now, we can create a function to define the :math:`i`\ th layer, acting
+# on qumode ``q``. This allows us to simply call this function in a loop
+# later on when we build our circuit.
+
+# layer architecture
+@operation(1)
+def layer(i, q):
+    Rgate(sf_params[i][0]) | q
+    Sgate(sf_params[i][1], sf_params[i][2]) | q
+    Rgate(sf_params[i][3]) | q
+    Dgate(sf_params[i][4], sf_params[i][5]) | q
+    Kgate(sf_params[i][6]) | q
+    return q
+
+
+######################################################################
+# Constructing the circuit
+# ------------------------
+#
+# Now that we have defined our gate parameters and our layer structure, we
+# can construct our variational quantum circuit.
+
+
+# Apply circuit of layers with corresponding depth
+with prog.context as q:
+    for k in range(depth):
+        layer(k) | q[0]
+
+
+######################################################################
+# Performing the optimization
+# ---------------------------
+#
+# :math:`\newcommand{ket}[1]{\left|#1\right\rangle}` With the Strawberry
+# Fields TensorFlow backend calculating the resulting state of the circuit
+# symbolically, we can use TensorFlow to optimize the gate parameters to
+# minimize the cost function we specify. With state learning, the measure
+# of distance between two quantum states is given by the fidelity of the
+# output state :math:`\ket{\psi}` with some target state
+# :math:`\ket{\psi_t}`. This is defined as the overlap between the two
+# states:
+#
+# .. math::  F = \left|\left\langle{\psi}\mid{\psi_t}\right\rangle\right|^2
+#
+# where the output state can be written
+# :math:`\ket{\psi}=U(\vec{\theta})\ket{\psi_0}`, with
+# :math:`U(\vec{\theta})` the unitary operation applied by the variational
+# quantum circuit, and :math:`\ket{\psi_0}=\ket{0}` the initial state.
+#
+# Let's define the target state as the single photon state
+# :math:`\ket{\psi_t}=\ket{1}`:
+
+import numpy as np
+
+target_state = np.zeros([cutoff])
+target_state[1] = 1
+print(target_state)
+
+
+######################################################################
+# Using this target state, we calculate the fidelity with the state
+# exiting the variational circuit. We must use TensorFlow functions to
+# manipulate this data, as well as a ``GradientTape`` to keep track of the
+# corresponding gradients!
+#
+# We choose the following cost function:
+#
+# .. math:: C(\vec{\theta}) = \left| \langle \psi_t \mid U(\vec{\theta})\mid 0\rangle - 1\right|
+#
+# By minimizing this cost function, the variational quantum circuit will
+# prepare a state with high fidelity to the target state.
```

Comment:
(Minor suggestion): or comparable such that maximizing is included again after the intro
```suggestion
# prepare a state with high fidelity to the target state (hence we achieve maximization).
```

Reply:
I'm not sure this sounds entirely right :thinking: